aten/src/ATen/templates/TensorBody.h

```diff
@@ -204,9 +204,17 @@ class CAFFE2_API Tensor {
     return impl_->is_non_overlapping_and_dense();
   }
 
-  at::MemoryFormat suggest_memory_format() const {
-    if (!is_mkldnn() && !is_sparse() && !impl_->is_contiguous() && impl_->is_strides_like_channels_last()) {
-      return at::MemoryFormat::ChannelsLast;
+  at::MemoryFormat suggest_memory_format(
```

Comment:
I'm not quite uncertain that I understand the behavior of `suggest_memory_format` here. I would prefer to do this right in a separate PR (add tests to define the proper behavior), so hopefully when Vitaly comes back it would be easier to communicate it through.